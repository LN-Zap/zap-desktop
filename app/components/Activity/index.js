import Activity from './Activity'

export default Activity
