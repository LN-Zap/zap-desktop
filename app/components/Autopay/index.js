import Autopay from './Autopay'

export default Autopay
export AutopayActions from './AutopayActions'
export AutopayCardView from './AutopayCardView'
export AutopayHeader from './AutopayHeader'
export AutopayHeading from './AutopayHeading'
export AutopayList from './AutopayList'
export AutopaySearch from './AutopaySearch'
export AutopaySearchNoResults from './AutopaySearchNoResults'
