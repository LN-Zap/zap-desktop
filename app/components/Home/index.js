export Home from './Home'
export CreateWalletButton from './CreateWalletButton'
export NoWallets from './NoWallets'
export WalletsMenu from './WalletsMenu'
export WalletLauncher from './WalletLauncher'
export WalletUnlocker from './WalletUnlocker'
export WalletSettingsFormLocal from './WalletSettingsFormLocal'
export WalletSettingsFormRemote from './WalletSettingsFormRemote'
export WalletHeader from './WalletHeader'
