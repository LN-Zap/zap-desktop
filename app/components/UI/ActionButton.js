import React, { useState, useEffect, useRef } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import delay from '@zap/utils/delay'
import Button from './Button'
import Spinner from './Spinner'

const StyledButton = styled(Button)`
  color: ${props => (props.active ? props.theme.colors.lightningOrange : null)};
  &:hover {
    color: ${props => props.theme.colors.lightningOrange};
  }
`
StyledButton.propTypes = {
  active: PropTypes.bool,
}

const ActionButton = ({ children, hint, onClick, timeout, ...rest }) => {
  const [status, setStatus] = useState(null)
  const buttonRef = useRef()

  async function triggerAction() {
    await onClick()
    await delay(timeout)
    setStatus('done')
  }

  useEffect(() => {
    if (status === 'fetching') {
      triggerAction()
    } else if (status === 'done') {
      buttonRef.current.blur()
    }
  }, [status])

  function handleClick() {
    setStatus('fetching')
  }

  return (
    <StyledButton
      ref={buttonRef}
      active={status === 'fetching'}
      className="hint--bottom-left"
      data-hint={hint}
      onClick={handleClick}
      size="small"
      variant="secondary"
      {...rest}
    >
      {status === 'fetching' ? <Spinner /> : children}
    </StyledButton>
  )
}

ActionButton.propTypes = {
  children: PropTypes.node,
  hint: PropTypes.node,
  onClick: PropTypes.func.isRequired,
  timeout: PropTypes.number,
}

ActionButton.defaultProps = {
  timeout: 1000,
}

export default ActionButton
