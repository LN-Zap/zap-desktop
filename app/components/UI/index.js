export AmountInput from './AmountInput'
export BackgroundPrimary from './BackgroundPrimary'
export BackgroundTertiary from './BackgroundTertiary'
export BackgroundSecondary from './BackgroundSecondary'
export Bar from './Bar'
export Button from './Button'
export Countdown from './Countdown'
export CryptoAmountInput from './CryptoAmountInput'
export DataRow from './DataRow'
export Dropdown from './Dropdown'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export Message from './Message'
export GlobalStyle from './GlobalStyle'
export Header from './Header'
export Heading from './Heading'
export Input from './Input'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export MainContent from './MainContent'
export Menu from './Menu'
export MenuItem from './MenuItem'
export MenuItemGroup from './MenuItemGroup'
export Modal from './Modal'
export Notification from './Notification'
export Panel, { PanelHeader, PanelBody, PanelFooter } from './Panel'
export Page from './Page'
export PasswordInput from './PasswordInput'
export QRCode from './QRCode'
export Radio from './Radio'
export RadioGroup from './RadioGroup'
export Range from './Range'
export Select from './Select'
export Sidebar from './Sidebar'
export Span from './Span'
export Spinner from './Spinner'
export Text from './Text'
export TextArea from './TextArea'
export Titlebar from './Titlebar'
export Toggle from './Toggle'
export Truncate from './Truncate'
export Value from './Value'
export Wizard from './Wizard'
