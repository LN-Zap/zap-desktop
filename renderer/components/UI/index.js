export ActionBar from './ActionBar'
export ActionButton from './ActionButton'
export BackgroundPrimary from './BackgroundPrimary'
export BackgroundSecondary from './BackgroundSecondary'
export BackgroundTertiary from './BackgroundTertiary'
export Bar from './Bar'
export { BasicInput } from './Input'
export { BasicRange } from './Range'
export { BasicTextArea } from './TextArea'
export { BasicCheckbox } from './Checkbox'
export { BasicSelect } from './Select'
export Button from './Button'
export Card from './Card'
export Checkbox from './Checkbox'
export CloseButton from './CloseButton'
export Countdown from './Countdown'
export CryptoAmountInput from './CryptoAmountInput'
export CurrencyFieldGroup from './CurrencyFieldGroup'
export DataRow from './DataRow'
export Dialog from './Dialog'
export DialogOverlay from './DialogOverlay'
export Donut from './Donut'
export Dropdown from './Dropdown'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export GlobalStyle from './GlobalStyle'
export Header from './Header'
export Heading from './Heading'
export Input from './Input'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export Link from './Link'
export LndConnectionStringInput from './LndConnectionStringInput'
export LndConnectionStringEditor from './LndConnectionStringEditor'
export MainContent from './MainContent'
export Message from './Message'
export Modal from './Modal'
export { ModalOverlayStyles } from './Modal'
export NodePubkeyInput from './NodePubkeyInput'
export Notification from './Notification'
export OpenDialogButton from './OpenDialogButton'
export OpenDialogInput from './OpenDialogInput'
export Page from './Page'
export Panel, { PanelHeader, PanelBody, PanelFooter } from './Panel'
export ProgressBar from './ProgressBar'
export QRCode from './QRCode'
export Radio from './Radio'
export RadioCard from './RadioCard'
export RadioGroup from './RadioGroup'
export Range from './Range'
export RowLabel from './RowLabel'
export Select from './Select'
export Sidebar from './Sidebar'
export Span from './Span'
export Spinner from './Spinner'
export StatusIndicator from './StatusIndicator'
export Tab from './Tab'
export Tabs from './Tabs'
export Text from './Text'
export TextArea from './TextArea'
export Titlebar from './Titlebar'
export Toggle from './Toggle'
export Tooltip from './Tooltip'
export TransactionFeeInput from './TransactionFeeInput'
export Value from './Value'
export Wizard from './Wizard'
