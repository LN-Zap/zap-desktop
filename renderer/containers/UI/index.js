export CurrencyFieldGroup from './CurrencyFieldGroup'
export CryptoSelector from './CryptoSelector'
export CryptoValue from './CryptoValue'
export FiatSelector from './FiatSelector'
export FiatValue from './FiatValue'
