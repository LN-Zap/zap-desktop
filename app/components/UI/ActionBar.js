import React from 'react'
import PropTypes from 'prop-types'
import { Flex } from 'rebass'
import { Button } from 'components/UI'

const createButtons = buttons =>
  buttons.map((entry, index) => (
    <Button
      key={entry.name}
      variant={index === buttons.length - 1 ? 'normal' : 'secondary'}
      mr={index < buttons.length - 1 ? 6 : 0}
      onClick={entry.onClick}
    >
      {entry.name}
    </Button>
  ))

const ActionBar = ({ isOpen, buttons, ...rest }) => {
  // check if buttons is a descriptive array  or a React renderable node
  const buttonsLayout = React.isValidElement(buttons) ? buttons : createButtons(buttons)

  return (
    <Flex
      flexDirection="row"
      justifyContent="flex-end"
      bg="secondaryColor"
      pr={5}
      pt={2}
      pb={2}
      {...rest}
    >
      {buttonsLayout}
    </Flex>
  )
}

ActionBar.propTypes = {
  buttons: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string.isRequired,
        onClick: PropTypes.func.isRequired
      })
    )
  ]).isRequired
}

export default ActionBar
