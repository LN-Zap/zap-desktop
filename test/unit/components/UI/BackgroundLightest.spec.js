import React from 'react'

import { renderWithTheme } from '@zap/test/unit/__helpers__/renderWithTheme'
import { BackgroundSecondary } from 'components/UI'

describe('component.UI.BackgroundSecondary', () => {
  it('should render correctly', () => {
    const tree = renderWithTheme(<BackgroundSecondary />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
