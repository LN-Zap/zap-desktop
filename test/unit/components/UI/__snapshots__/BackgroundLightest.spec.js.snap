// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.BackgroundSecondary should render correctly 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: secondaryColor;
  color: primaryText;
}

<div
  className="c0"
  color="primaryText"
/>
`;
